### PR TITLE
Offboard an AMC organization from edX

### PR DESCRIPTION
**Pull request by ahmedaljazzar**
## Background
We started this PR earlier in 2019 to help us remove a specific site from Tahoe. This step is necessary when we need to offboard a customer. However, we didn't have such a case until nw when one of our clients started working on their disaster and recovery plan. 
What we are trying to achieve here, is building a tool that removes all high-level data of a specific organization. This data include high level hooks and relations (No courses, material, or logs):
- Microsite Object.
- Organization Object.
- Tier Object

## This PR
Here we offboard an organization through a new API from edX platform by removing its data from the system. I also fixed the broken `delete_site` function that's now being used to offboard organizations.

I was thinking of keeping all deleted organizations in the database, instead of removing them we can add a new field like `is_active` or similar, but that at the end of the day depends on the business logic, the customers we have, and how deep we are planning to go.

## Steps to Reproduce
To be able to use the API, you need to
- SSH into your LMS shell; on devstack you can use `make lms-shell`.
- Run the management command `python manage.py lms <domain>. If you are running this on a local machine your domain should be `<slug>.localhost:18000`

Note: you can now create a website using `python manage.py lms create_devstack_site <slug>`
## Related PRs
- [amc#197](https://github.com/appsembler/amc/pull/197)

### CODE REVIEW

**Comment by johnbaldwin** [comment]
@ahmedaljazzar Who would be performing the call? A customer or us? cc @OmarIthawi

**Comment by OmarIthawi** [comment]
@johnbaldwin 

> @ahmedaljazzar Who would be performing the call? A customer or us? cc @OmarIthawi

A Tahoe infrastructure admin like me or @ahmedaljazzar

File: openedx/core/djangoapps/appsembler/sites/api.py [hunk 1]
```diff
@@ -66,6 +66,17 @@ def get_serializer_class(self):
...
+    def perform_destroy(self, instance):
+        delete_site(instance.site)
+
+
+class OffboardOrganizationAPIView(generics.DestroyAPIView):
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (IsAuthenticated, AMCAdminPermission)
+    queryset = Site.objects.all()
+    serializer_class = SiteSerializer
+    lookup_field = 'domain'
```
[the remark this answers is not in the document]
Comment (ahmedaljazzar) [reply openedx/core/djangoapps/appsembler/sites/api.py]: Yes. The primary key for Django's Site model is the domain.

**Comment by ahmedaljazzar** [comment]
> @ahmedaljazzar Who would be performing the call? A customer or us? cc @OmarIthawi

@johnbaldwin every one who have SSH access to our infrastructure.

**Comment by ahmedaljazzar** [comment]
@OmarIthawi This is ready for another pass.

**Comment by ahmedaljazzar** [comment]
@OmarIthawi This is also ready for another pass.

**Comment by OmarIthawi** [comment]
Same here, thank you again!

File: openedx/core/djangoapps/appsembler/sites/management/commands/remove_site.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        self.stdout.write('Removing "%s" in progress...' % organization_domain)
+        organization = self._check_orrganization(organization_domain)
+
+        delete_site(organization)
+        self.stdout.write(self.style.SUCCESS('Successfully removed site "%s"' % organization_domain))
+
+    def _check_orrganization(self, domain):
+        """
+        Locates the organization to be deleted and return its instance.
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/sites/management/commands/remove_site.py]:
```suggestion
        organization = self.get_site(organization_domain)

        delete_site(organization)
        self.stdout.write(self.style.SUCCESS('Successfully removed site "%s"' % organization_domain))

    def get_site(self, domain):
        """
        Locates the site to be deleted and return its instance.
```

**Comment by ahmedaljazzar** [comment]
Applied the function renaming request, and squashed the commits. Waiting for tests to pass to merge.

**Pull request merged**